agent/agentcontainers/subagent.go

```diff
@@ -233,19 +233,27 @@ func (a *subAgentAPIClient) Create(ctx context.Context, agent SubAgent) (SubAgen
 	if err != nil {
 		return SubAgent{}, err
 	}
+	defer func() {
+		if err != nil {
+			// Best effort.
+			_, _ = a.api.DeleteSubAgent(ctx, &agentproto.DeleteSubAgentRequest{
+				Id: resp.GetAgent().GetId(),
+			})
+		}
+	}()
 
-	agent.Name = resp.Agent.Name
-	agent.ID, err = uuid.FromBytes(resp.Agent.Id)
+	agent.Name = resp.GetAgent().GetName()
+	agent.ID, err = uuid.FromBytes(resp.GetAgent().GetId())
 	if err != nil {
-		return agent, err
+		return SubAgent{}, err
 	}
-	agent.AuthToken, err = uuid.FromBytes(resp.Agent.AuthToken)
+	agent.AuthToken, err = uuid.FromBytes(resp.GetAgent().GetAuthToken())
 	if err != nil {
-		return agent, err
+		return SubAgent{}, err
```

Comment:
Review: In either this or above, we could have returned a semi-valid agent and assigned to `proc.agent` was is not ideal.